source/common/http/http1/balsa_parser.cc

```diff
@@ -103,6 +104,17 @@ bool isUrlValid(absl::string_view url, bool is_connect) {
          std::all_of(path_query.begin(), path_query.end(), is_valid_path_query_char);
 }
 
+bool isVersionValid(absl::string_view version_input) {
+  // HTTP-version is defined at
+  // https://www.rfc-editor.org/rfc/rfc7230.html#section-2.6. HTTP/0.9 requests
+  // have no http-version, so empty `version_input` is also accepted.
+  envoy::type::matcher::v3::RegexMatcher matcher;
+  *matcher.mutable_google_re2() = envoy::type::matcher::v3::RegexMatcher::GoogleRE2();
+  matcher.set_regex("|HTTP/[0-9]\\.[0-9]");
+  const auto regex = Regex::Utility::parseRegex(matcher);
```

Comment:
I think this will compile a regex object every time this function is called. Is there any chance we could use a static const regex object?

Reply:
Ah, great catch.  Done!